### ISSUE TITLE
MaxMessageLength

### ISSUE THREAD

**Issue by farhodyusupov** [grpc/grpc-dart#659]
<short description>
Is there any way to set up MaxMessageLength?
<version of the grpc-dart packages used; see your `pubspec.lock` file>

## Repro steps

1. step1
1. step2
1. step3

Expected result: <describe what should have happened>

Actual result: <describe what actually happened>

## Details

<Include any other relevant details, logs, etc.>

**Comment by mosuem** [comment]
No, but please feel free to file a PR to implement this in `ChannelOptions`.

**Comment by farhodyusupov** [comment]
can you suggest something to start with?

**Comment by mosuem** [comment]
@jtattermusch I couldn't find any specs for this feature. Do specs for this exist, or is there a reference implementation?

**Comment by mraleph** [comment]
@mosuem I think this is about configuring `CodedBufferReader`'s [`sizeLimit`](https://github.com/google/protobuf.dart/blob/master/protobuf/lib/src/protobuf/coded_buffer_reader.dart#L25).

**Comment by jtattermusch** [comment]
> @jtattermusch I couldn't find any specs for this feature. Do specs for this exist, or is there a reference implementation?

This is what's done for C-core: https://github.com/grpc/grpc/blob/12c97481349d88852490a4acd0b5a0f9301d5b6c/include/grpc/impl/channel_arg_names.h#L37
https://github.com/grpc/grpc/blob/12c97481349d88852490a4acd0b5a0f9301d5b6c/include/grpc/impl/channel_arg_names.h#L43

grpc would have something similar.

**Comment by farhodyusupov** [comment]
thank you, If I could succeed I will send pull request

**Comment by farhodyusupov** [comment]
> @mosuem I think this is about configuring `CodedBufferReader`'s [`sizeLimit`](https://github.com/google/protobuf.dart/blob/master/protobuf/lib/src/protobuf/coded_buffer_reader.dart#L25).

I didn`t work
